include/swift/AST/SimpleRequest.h

```diff
@@ -318,7 +318,8 @@ class SimpleRequest<Derived, Output(Inputs...), Caching> {
   }
 
   void diagnoseCycle(DiagnosticEngine &diags) const {
-    diags.diagnose(asDerived().getNearestLoc(), diag::circular_reference);
+    diags.diagnose(asDerived().getNearestLoc(), diag::circular_reference,
+                   TypeID<Derived>::getName());
```

Comment:
This is a bit too low-level. It would take a bit of work, but we should add a space-separated lower-case name to each request too, and then use that